src/render/tutorials/graphql-usage/2_0.html.md

```diff
@@ -0,0 +1,777 @@
+    }
+}
+
+exposeInPayload-getBook {
+    factory = expose-payload-data
+    config {
+        key = getBook
+        exposeAs = fetchedData
+    }
+}
+```
+
+And then setup transitions in our tasks. They should now look like this:
+
+```HOCON
+tasks {
+  get-books {
+    action = getBooks
+    onTransitions {
+      _success {
+        action = exposeInPayload-getBooks
+      }
+    }
+  }
+
+  get-book {
+    action = getBook
+    onTransitions {
+      _success {
+        action = exposeInPayload-getBook
+      }
+    }
+  }
+}
+```
+
+## We're done
+
+That's it. You can now run the application. Easiest way of seeing it in action is using postman, but you can use curl or anything else you want.
+
+Let's construct a sample query:
+
+```graphql
+{
+    book(id: "q5NoDwAAQBAJ") {
+        title
+        authors
+    }
+    books(match: "java") {
+        publisher
+    }
+}
+```
+
+We're asking for the title and authors of the book with id "q5NoDwAAQBAJ" and a list of books about java with only their publishers listed. Request it at http://localhost:8092/api/graphql
+
+If you're using postman you can just choose predefined `GraphQL` body type and paste the query there. If you prefer curl:
+
+```bash
+curl -i -H 'Content-Type: application/json' -X POST -d '{"query": "{book(id: \"q5NoDwAAQBAJ\") {title authors} books(match: \"java\") {publisher}}"}' http://localhost:8092/api/graphql
+```
+
+You should get the following response:
+
+```json5
+{
+    "data": {
+        "book": {
+            "title": "Learning GraphQL",
+            "authors": [
+                "Eve Porcello",
+                "Alex Banks"
+            ]
+        },
+        "books": [
+            {
+                "publisher": "Helion"
+            },
+            {
+                "publisher": "Helion"
+            },
+            {
+                "publisher": "Helion"
+            },
+            {
+                "publisher": "\"O'Reilly Media, Inc.\""
+            },
+            {
+                "publisher": "Morgan Kaufmann"
+            },
+            //...
+        ]
+    }
+}
+```
+
+## Few notes at the end
+As you can see Knot.x is highly customizable. Data fetchers that we implemented are quite generic. You can use them to easily create more advanced GraphQL APIs without the need for writing more fetchers. For clarity purposes the code in this tutorial don't have advanced error handling and is quite error-prone. It's left as an exercise for you to implement better error handling.
+
+Also, you might have noted that our `fromJson` method in `GraphQLDataObject` interface have a `DataFetchingEnvironment` argument that we didn't use. This is because our example is very simple, but the argument is included in code to show the flexibility of our solution. You can imagine a more complex case where we need to parse json to model differently, dependending on query parameters. This is where we could use our `DataFetchingEnvironment` argument.

```

Comment:
```suggestion
Also, you might have noted that our `fromJson` method in `GraphQLDataObject` interface has a `DataFetchingEnvironment` argument that we didn't use. This is because our example is very simple, but the argument is included in code to show the flexibility of our solution. You can imagine a more complex case where we need to parse JSON to model differently, depending on query parameters. This is where we could use our `DataFetchingEnvironment` argument.
```